src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1454,7 +1454,13 @@ bool isContainingAssemblyInReferences(ISymbol s)
                             // a containing assembly, we treat them as in the current assembly for access purposes
                             return assemblyIsInReferences(s.ContainingAssembly ?? this.Assembly);
                         default:
-                            return assemblyIsInReferences(s.ContainingAssembly);
+                            var containingAssembly = s.ContainingAssembly;
+                            if (containingAssembly is null)
+                            {
+                                Debug.Assert(false);
```

Comment:
Consider passing a message to `Debug.Assert` indicating why containingAssembly is expected to be non-null. So it's more actionable in future if the assertion happened to fail.